README.md

```diff
@@ -0,0 +1,560 @@
+
+### 4. Mobile-services parser
+The Mobile services parser is a helper class to parse the `mobile-services.json` file. 
+This file contains the connection details for the services used within the application.
+
+`IMobileService` has method stubs for configuration getters required to configure the `Apollo-Client`, `App-Auth` and `Unified push`. 
+These methods are:
+
+- `String getGraphqlServer()` = Return the URL for a GraphQL server.
+- `String getKIssuer()` = Returns the Issuer URL for AppAuth.
+- `String getKClientId()` = Returns the client Id for AppAuth 
+- `String getPushUrl()` = Returns the URL for the unified push server.
+- `String getPushVariantId()` = Returns the Variant ID for the unified push server.
+- `String getPushVariantSecret()` = Returns the Variant Secret for the unified push server.
+ 
+#### mobile-services.json
+ 
+In our example, `mobile-services.json` is saved in `app/src/main/assets/config`.
+There are three types of services in the services list. 
+Each of these have the configuration for the different type services which are:
+
+- **sync-app** 
+- **keycloak**
+- **push** 
+
+##### sync-app
+Sync-app holds the configuration for connecting to a GraphQL server.
+The JSON object in the services list, at minimum, needs the fields in the example below.
+
+Some GraphQL servers require a web socket URL which is placed in the `config` object.
+When this object is converted into a java object, it takes the form of `SyncApp.class`
+
+```json
+  {
+    "config": {
+      "websocketUrl": "wss://server.example.com/graphql"
+    },
+    "name": "sync-app",
+    "type": "sync-app",
+    "url": "https://server.example.com/graphql"
+  }
+```
+##### Keycloak
+
+The JSON object example below holds the configuration for Keycloak server which would be in the services array.
+When the `mobile-service.json` file is parsed, configuration build is converted to a java `Keycloak.class` object.
+
+```json
+    {
+      "config": {
+        "auth-server-url": "https://sso.example.com/auth",
+        "confidential-port": 0,
+        "public-client": true,
+        "realm": "example-app-realm",
+        "resource": "example-app-client",
+        "ssl-required": "external"
+      },
+      "name": "keycloak",
+      "type": "keycloak",
+      "url": "https://sso.example.com/auth"
+    }
+```
+##### Push
+
+The Push example configuration is not used in our example, but the following would be its format.
+
+```json
+    {
+      "config": {
+        "android": {
+          "variantID": "variantID (e.g. 1234456-234320)",
+          "variantSecret": "variantSecret (e.g. 1234456-234320)"
+        }
+      },
+      "name": "push",
+      "type": "push",
+      "url": "https://push.example.com"
+    }
+```
+The only information that is missing is `sender Id`, which can be found in application settings on firebase, or in the `google-services.json` file, which is required in this example application.
+
+#### MobileService.class
```

Comment:
we can keep this section and reduce the content a bit perhaps. Just need to explain how to get configurations.